code/game/gamemodes/management/event/combat.dm

```diff
@@ -11,8 +11,21 @@
 
 /datum/game_mode/combat/post_setup()
 	..()
+	//No more OOC
+	GLOB.ooc_allowed = FALSE
+	to_chat(world, "<B>Due to gamemode, the OOC channel has been globally disabled.</B>")
+	to_chat(world, "<span class='userdanger'>Round will end in 40 minutes.</span>")
+
+	//Breach all
 	for(var/mob/living/simple_animal/hostile/abnormality/A in GLOB.mob_list)
 		INVOKE_ASYNC(A, /mob/living/simple_animal/hostile/abnormality.proc/BreachEffect)
 		var/obj/effect/proc_holder/spell/targeted/night_vision/bloodspell = new
 		A.AddSpell(bloodspell)
 		A.faction += "hostile"
+	if(SSmaptype.maptype in SSmaptype.autoend)
+		addtimer(CALLBACK(src, .proc/loseround), 40 MINUTES)
+
+/datum/game_mode/combat/proc/loseround()
+	SSticker.force_ending = 1
+	for(var/mob/M in GLOB.player_list)
+		to_chat(M, "<span class='userdanger'>Players have taken too long! Round automatically ending.</span>")
```

Comment:
```suggestion
	for(var/mob/M in GLOB.player_list)
		var/text_class = "'userdanger'"
		if("hostile" in M.faction)
			text_class = "'nicegreen'"
		to_chat(M, "<span class="+text_class+">Rabbits have failed to initiate site burial in time and any remaining abnormalities have otherwise escaped! Round automatically ending.</span>")
```

Reply:
Okay so
The thing is that it's not actually just for this one thing, it's also used for Limbus Company 13.